lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -131,67 +126,120 @@ function dosomething_kudos_sort_by_taxonomy_term($a, $b) {
   return $result;
 }
 
+/**
+ * Get current user from supplied data array or global user.
+ *
+ * @param  array  $data
+ * @return null|string
+ */
+function dosomething_kudos_get_current_user($data = []) {
```

Comment:
hrm two things
1. this seems more like a global or helper function
2. it can/probably should be done in way less checks